### PR TITLE
[Seq] Remove incorrect canonicalization

### PR DESCRIPTION
**Pull request by uenoku**
Fix https://github.com/llvm/circt/issues/7266

### CODE REVIEW

File: lib/Dialect/Seq/SeqOps.cpp [hunk 1]
```diff
@@ -577,9 +577,8 @@ LogicalResult FirRegOp::canonicalize(FirRegOp op, PatternRewriter &rewriter) {
   };
 
   if (isConstant()) {
```
Comment (dtzSiFive) [inline lib/Dialect/Seq/SeqOps.cpp:579]: `if (isConstant() && !op.getResetValue()) {`
?

File: test/Dialect/Seq/canonicalization.mlir [hunk 2]
```diff
@@ -42,7 +42,7 @@ hw.module @FirRegReset(in %clk : !seq.clock, in %in : i32, in %r : i1, in %v : i
   // value.
```
Comment (dtzSiFive) [inline test/Dialect/Seq/canonicalization.mlir:42]: Please update the comment / test.

Just curious, does this have an impact on our designs in practice?

**Review by dtzSiFive — APPROVED** [review]
Thanks for fixing this.  I think LGTM but I think there's more work to do here.

Probably can just simplify this logic now (see comment)?

---

Not for this PR, but when looking at this myself earlier:

1) What role should preset values have re:folders/canonicalization? Not really asking for an answer but it does seem our folders/canonicalizers should be updated to handle that.
2) Folder and canonicalizer are not aligned.  A firreg with "trivial feedback" with a constant zero reset and a constant reset value-- canonicalizer sees constant zero reset and drops the reset and reset value, folder will replace it with a constant zero.  If folder goes first it replaces the register with the constant reset value.

Here's some IR for (2) that acts as described (point `circt-opt -canonicalize` at it, you get `123`, disable that part of the folder and it'll become zero!).

```mlir
hw.module.extern @Observe(in %x : i32)

hw.module @FirRegReset(in %clk : !seq.clock, in %r : i1) {
  %false = hw.constant false
  %constant = hw.constant 123 : i32

  %reg = seq.firreg %reg clock %clk reset sync %false, %constant : i32
  hw.instance "sink" @Observe(x: %reg: i32) -> ()
}
```

It would be good for us to sit down and revisit the justification and correctness for these.

File: lib/Dialect/Seq/SeqOps.cpp (hunk 1 above)
Comment (uenoku) [reply lib/Dialect/Seq/SeqOps.cpp]: Yes, thanks!

**Comment by uenoku** [comment]
> 1. What role should preset values have re:folders/canonicalization? Not really asking for an answer but it does seem our folders/canonicalizers should be updated to handle that.

Yeah good point. I agree we need to update canonicalizations. For now I added a condition to stop fold/canonicalization for a register with a preset value. Let's revisit later.

File: test/Dialect/Seq/canonicalization.mlir (hunk 2 above)
Comment (uenoku) [reply test/Dialect/Seq/canonicalization.mlir]: There was no change in large design 🤔

**Comment by dtzSiFive** [comment]
Thanks, @uenoku !! 🚀

**Pull request merged**